docs/DeveloperGuide.md

```diff
@@ -250,61 +250,224 @@ Priorities: High (must have) - `* * *`, Medium (nice to have)  - `* *`,  Low (un
+
+3b. The command format is invalid
+> 3b1. Task!t shows a 'invalid command' message and display the expected format.<br>
+> Use case resumes at step 2
 
+#### Use case: View task
+
+**MSS**
+
+1. User requests to view tasks for a particular day/week/month/date
+2. Task!t display the list of tasks for that date sorted by date and/or time<br>
+Use case ends.
+
+**Extensions**
+
+2a. The list is empty
+> 2a1. Task!t shows a 'no task found' message.<br>
+> Use case ends
+
+#### Use case: Find task
+
+**MSS**
+
+1. User requests to find tasks by keyword
+2. Task!t displays the list of tasks that contains the keyword in the name<br>
+Use case ends.
+
+**Extensions**
+
+2a. The list is empty
+> 2a1. Task!t shows a 'no task found' message.<br>
+> Use case ends
+
+#### Use case: Mark task as done
+
+**MSS**
+
+1. User requests to view tasks for a particular date/week/month/date
+2. Task!t displays the list of tasks for that date sorted by date and/or time
+3. User request to mark specific tasks in the list based on task’s index
+4. Task!t marks the tasks<br>
+Use case ends.
+
+**Extensions**
+
+2a. The list is empty
+> 2a1. Task!t shows a 'no task found' message.<br>
+> Use case ends
+
+3a. The given index is invalid
+> 3a1. Task!t shows a 'invalid index' message <br>
+> Use case resumes at step 2
+
+3b. The command format is invalid
+> 3b1. Task!t shows a 'invalid format' message and display the expected format<br>
+> Use case ends
+
+3c. The task is already marked as done
+> 3c1. Task!t shows a 'task already marked' message.<br>
+> Use case resumes at step 3
+
+#### Use case: Delete task
+
+**MSS**
+
+1. User requests to view tasks for a particular date/week/month/date
+2. Task!t displays the list of tasks for that date sorted by date and/or time
+3. User requests to delete a specific task in the list based on task’s index
+4. Task!t deletes the task <br>
+Use case ends.
+
+**Extensions**
+
+2a. The list is empty
+> 2a1. Task!t shows a 'no task found' message<br>
 > Use case ends
 
 3a. The given index is invalid
+> 3a1. Task!t shows a 'invalid index' message <br>
+> Use case resumes at step 2
+
+#### Use case: Set storage location
 
-> 3a1. AddressBook shows an error message <br>
-  Use case resumes at step 2
+**MSS**
+
+1. User requests to change storage location wo a specified file path
+2. Task!t changes the storage location<br>
+Use case ends.
+
+**Extensions**
 
-{More to be added}
+2a. The given path is invalid
+> 2a1. Task!t shows a 'invalid path' message<br>
+> Use case ends
 
 ## Appendix C : Non Functional Requirements
 
 1. Should work on any [mainstream OS](#mainstream-os) as long as it has Java `1.8.0_60` or higher installed.
-2. Should be able to hold up to 1000 persons.
+2. Should be able to hold up to 1000 tasks.
 3. Should come with automated unit tests and open source code.
 4. Should favor DOS style commands over Unix-style commands.
-
-{More to be added}
+5. Should display response within 1 second.
+6. Should work stand-alone.
```

Comment:
The description is quite abstract for this NFR. Please add more information to explain what you mean by `stand-alone`.